### PR TITLE
[KYUUBI #4920] Add OperationRestApi

### PR DESCRIPTION
**Pull request by lightning-L**
### _Why are the changes needed?_

close https://github.com/apache/kyuubi/issues/4920

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by lightning-L** [comment]
to close https://github.com/apache/kyuubi/issues/4602

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/OperationsResource.scala [hunk 1]
```diff
@@ -81,6 +81,9 @@ private[v1] class OperationsResource extends ApiRequestContext with Logging {
       }
       Response.ok().build()
     } catch {
+      case e: KyuubiSQLException =>
+        error(e.getMessage, e)
+        throw e
```
Comment (turboFei) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/OperationsResource.scala:86]: also cc @pan3793

**Review by turboFei — APPROVED** [review]
LGTM

**Review by cxzl25 — APPROVED** [review]

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/OperationsResource.scala (hunk 1 above)
Comment (pan3793) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/OperationsResource.scala:86]: is that aim to address no error log issue? I also find the same issue and I think we should log it in `RestExceptionMapper` for all exceptions

Comment (pan3793) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/OperationsResource.scala:86]: Open #4977 to address it generally.

File: kyuubi-rest-client/src/main/java/org/apache/kyuubi/client/api/v1/dto/KyuubiOperationEvent.java [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.client.api.v1.dto;
+
+import java.util.Map;
+
+public class KyuubiOperationEvent implements KyuubiEvent {
```
Comment (pan3793) [inline kyuubi-rest-client/src/main/java/org/apache/kyuubi/client/api/v1/dto/KyuubiOperationEvent.java:22]: Do we need the interface KyuubiEvent? what's the relationship between the client-side KyuubiEvent and server-side KyuubiEvent

Comment (lightning-L) [reply kyuubi-rest-client/src/main/java/org/apache/kyuubi/client/api/v1/dto/KyuubiOperationEvent.java]: yes, we do not need the interface actually. Have removed it.

**Comment by pan3793** [comment]
Thanks, merged to master
